command/agentproxyshared/auth/cert/cert_test.go

```diff
@@ -188,4 +194,48 @@ func TestCertAuthMethod_AuthClient_withCertsReload(t *testing.T) {
 	if reloadedClient == clientToUse {
 		t.Fatal("expected client from AuthClient to return back a new client")
 	}
+
+	method.CredSuccess()
+	ctx, cancel := context.WithTimeout(context.TODO(), 3*time.Second)
+	inst := method.(*certMethod)
+	inst.clientCert = "./test-fixtures/keys/cert1.pem"
+	select {
+	case <-ctx.Done():
+	case <-method.NewCreds():
+		cancel()
+		t.Fatal("malformed cert should not be observed as a change")
+	}
+
+	inst.clientKey = "./test-fixtures/keys/key1.pem"
+	ctx, cancel = context.WithTimeout(context.TODO(), 3*time.Second)
+	select {
+	case <-ctx.Done():
+		t.Fatal("failed to watch the cert change: timeout")
+	case <-method.NewCreds():
+		cancel()
+	}
+}
+
+func TestHashCert(t *testing.T) {
```

Comment:
This should have a godoc explaining what the test does and what it's testing

Reply:
Hi, I split this unit test into 3 with inline docs and function doc to `hashCert()`. Please take a look if I need to add a more comprehensive on them. Thanks